### ISSUE TITLE
What is the appropriate accuracy of a pre-trained model?

### ISSUE THREAD

**Issue by yuyadanyadan** [MAGICS-LAB/DNABERT_2#122]
Hi, thank you for your working!  When I pre-train model on my own data by the following input.  The training data is the whole genome data of Chlorella, sliced into 1000 bases for one row of sequence; a total of 53,160 rows of data were generated, of which the first 80% of the 42,528 rows of sequence data was the training set, and the remaining 20% of the 10,632 rows of sequence data was the test set.
The run_mlm.py of hugging face was used from https://github.com/huggingface/transformers/tree/main/examples/pytorch/language-modeling. 

But  the accuracy of pre-trained model was only 0.1117.  How do I to improve the accuracy of pre-trained models? What is the accuracy of this pre-trained model for subsequent fine-tuning? 

![1](https://github.com/user-attachments/assets/3e3952a5-9285-4c1d-82d5-5f4bdf23d28b)
![2](https://github.com/user-attachments/assets/6f7905c7-f938-4fd1-863f-d9f1d7b18c28)

Thank you so much!

**Comment by Zhihan1996** [comment]
Thanks for using our work! First of all, the MLM accuracy is highly related to the vocabulary size and masking rate. Are you using DNABERT-2's tokenizer and a masking rate of 15%? In our case, the MLM accuracy in pre-training is around 30%.

**Comment by yuyadanyadan** [comment]
Thanks for your reply!  The input parameter "tokenizer_name" of run_mlm.py is the downloaded model of zhihan1996/DNABERT-2-117M    (/share/home/yuyadan/workspace/DNABERT_2/DNABERT-2-117M).  The masking rate is mlm_probability  which is default=0.15.  Should I modify the vocabulary size and mask rate to improve the accuracy of the pre-trained models?

**Comment by yuyadanyadan** [comment]
The config.json generated when I pre-trained the model was as shown below. I generated the config.json automatically by adding the parameter model_type and did not pass the DNABERT-2-117M model because of a conflict in the transformers version (DNABERT-2-117 requires a transformers 4.29, while run_mlm.py requires version 4.46). With the generated config.json, I found that the vocab_size is 30522 instead of 4096. how do I change this parameter, I didn't find the initialization of this parameter in run_mlm.py.

Translated with www.DeepL.com/Translator (free version)
![1](https://github.com/user-attachments/assets/37cc8019-b853-4e66-aab9-2da83e171e27)

**Comment by yuyadanyadan** [comment]
The config.json generated when I pre-trained the model was as shown below. I generated the config.json automatically by adding the parameter model_type and did not pass the DNABERT-2-117M model because of a conflict in the transformers version (DNABERT-2-117 requires a transformers 4.29, while run_mlm.py requires version 4.46). With the generated config.json, I found that the vocab_size is 30522 instead of 4096. how do I change this parameter, I didn't find the initialization of this parameter in run_mlm.py.
![1](https://github.com/user-attachments/assets/7ea05a13-56d7-4f0c-bdee-634fd6d800de)
